src/CMakeLists.txt

```diff
@@ -32,42 +32,38 @@ source_group("include_kea" FILES ${LIBKEA_H})
 add_library(${LIBKEA_LIB_NAME} ${LIBKEA_CPP} ${LIBKEA_H} )
 target_link_libraries(${LIBKEA_LIB_NAME} ${HDF5_LIBRARIES} ${HDF5_HL_LIBRARIES})
 
-if(BUILD_SHARED_LIBS)
-    SET_TARGET_PROPERTIES(${LIBKEA_LIB_NAME}
-            PROPERTIES
-            SOVERSION ${LIBKEA_VERSION_MAJOR}.${LIBKEA_VERSION_MINOR}
-            VERSION ${LIBKEA_VERSION}
-            CLEAN_DIRECT_OUTPUT 1)
-else()
-    SET_TARGET_PROPERTIES(${LIBKEA_LIB_NAME}
-            PROPERTIES
-            OUTPUT_NAME "${LIBKEA_LIB_NAME}"
-            PREFIX "lib"
-            VERSION ${LIBKEA_VERSION}
-            CLEAN_DIRECT_OUTPUT 1)
+include(GenerateExportHeader)
+generate_export_header(${LIBKEA_LIB_NAME}
+  EXPORT_MACRO_NAME KEA_EXPORT
+  EXPORT_FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/include/libkea/kea_export.h
+)
+target_include_directories(${LIBKEA_LIB_NAME} PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
+
+set_target_properties(${LIBKEA_LIB_NAME}
+  PROPERTIES
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN 1
+    PREFIX "lib"
+    SOVERSION ${LIBKEA_VERSION_MAJOR}.${LIBKEA_VERSION_MINOR}
+    VERSION ${LIBKEA_VERSION}
+)
+if(MSVC AND NOT BUILD_SHARED_LIBS)
+  set_target_properties(${LIBKEA_LIB_NAME} PROPERTIES OUTPUT_NAME "libkea")
```

Comment:
So the output is still `libkea.dll` on Windows? I really don't want to be changing the ABI at this stage...

Reply:
Yes.

On Windows:
- shared build:
  - dll filename will be `libkea.dll`.
  - import lib filename will be `libkea.lib` with MSVC and `libkea.dll.a` with MinGW.
- static build: filename will be `liblibkea.lib` with MSVC (like before, that's weird but didn't change that because I saw some hardcoded things related to filenames in CMakeLists of gdal driver), and `libkea.a` with MinGW.

Reply:
Good. Yes `liblibkea.lib` is weird, but don't want to change anything now that could break things. Maybe could tidy this up if other ABI changes are made in the future.

Reply:
On Windows the import library is `kea.lib` breaking existing build scripts (although the dll is `libkea.dll`). Can this be fixed so it matches the old behaviour?

Reply:
Good catch. `PREFIX` property doesn't add prefix do import lib of dll, but `IMPORT_PREFIX` does.

Reply:
fixed by https://github.com/ubarsc/kealib/pull/10/commits/80fdd781557b37eb30582d0f8dbb3f002d852a2a